Gems/LmbrCentral/Code/Source/Builders/LuaBuilder/LuaBuilderWorker.cpp

```diff
@@ -43,6 +54,42 @@ namespace LuaBuilder
                 {
                     AZStd::string watchFolder;
                     AZ::Data::AssetInfo assetInfo;
+#if defined(CARBONATED)
+                    AZ::IO::Path path(dependency.m_dependencyPath.starts_with("/") || dependency.m_dependencyPath.starts_with("\\")
+                        ? dependency.m_dependencyPath.substr(1) : dependency.m_dependencyPath); // path should not start from a path delimiter
+
+                    if (path.HasExtension())
+                    {
+                        const auto it = AssetExtensionReplacementMap.find(path.Extension().String());
+                        if (it != AssetExtensionReplacementMap.end())
+                        {
+                            path.ReplaceExtension(it->second.c_str());
+                        }
+
+                        if (!assetSystem->GetSourceInfoBySourcePath(path.c_str(), assetInfo, watchFolder))
+                        {
+                            AZ_Warning("LuaBuilder", false, "Did not find dependency '%s' referenced by script (or it is not an asset path).", dependency.m_dependencyPath.c_str());
+                        }
+                        else if (assetInfo.m_assetId.IsValid())
```

Comment:
Minor change here.  Clean up as follows:
```
bool success = assetSystem->GetSourceinfoBySourcePath(..)
if(success)
{
    if(assetInfo.. IsValid())
         ... 
    else
         ...
}
else 
   ...
```

It's easier to read - I'm not a fan of assetInfo being set (if successful) in the conditional of the precedening"if" before it.

Reply:
Updated as described above ^^^